smart-contract-verifier/smart-contract-verifier/src/verifier/all_metadata_extracting_verifier.rs

```diff
@@ -105,6 +107,212 @@ impl<T: Source> Verifier<T> {
                         local_bytecode,
                         match_type,
                     }) => {
+                        mod raw_types {
+                            use serde::Deserialize;
+                            use std::collections::BTreeMap;
+
+                            #[derive(Clone, Debug, Deserialize, PartialEq)]
+                            pub struct CompilerOutput {
+                                #[serde(default)]
+                                pub contracts: BTreeMap<String, BTreeMap<String, Contract>>,
+                            }
+
```

Comment:
I suggest to move logic of extraction artifacts to separate file, or to the bottom of this file.
in my opinion, `verify` function should not be that big and it should show more business logic and less implementation of parsing